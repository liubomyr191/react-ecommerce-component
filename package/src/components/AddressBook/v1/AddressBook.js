import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
// import styled from "styled-components";
import { withComponents } from "@reactioncommerce/components-context";
import { CustomPropTypes } from "../../../utils";

// onst StyledDiv = styled.div`
//   color: #333333;
// `;

class AddressBook extends Component {
  static propTypes = {
    /**
     * User account data.
     */
    account: PropTypes.object,
    /**
     * If you've set up a components context using @reactioncommerce/components-context
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({
      /**
       * Pass either the Reaction AddressForm component or your own component that
       * accepts compatible props.
       */
      AddressForm: CustomPropTypes.component.isRequired,
      /**
       * Pass either the Reaction AddressReview component or your own component that
       * accepts compatible props.
       */
      AddressReview: CustomPropTypes.component.isRequired,
      /**
       * Pass either the Reaction AddressSelect component or your own component that
       * accepts compatible props.
       */
      AddressSelect: CustomPropTypes.component.isRequired
    }).isRequired,
    /**
     * Value for the AddressFrom
     * needed for editing anonymous cart shipping addresses
     */
    value: PropTypes.object
  };

  static defaultProps = {};

  state = {};

  _addresForm = null;
  _addressSelect = null;
  _addressReview = null;

  //
  // Handler Methods
  //

  //
  // Render Methods
  //

  renderAddressSelect() {
    const { components: { AddressSelect } } = this.props;
    return (
      <AddressSelect
        ref={(el) => {
          this._addressSelect = el;
        }}
      />
    );
  }

  renderAddressReview() {
    const { components: { AddressReview } } = this.props;
    return (
      <AddressReview
        ref={(el) => {
          this._addressReview = el;
        }}
      />
    );
  }

  renderAddressForm() {
    const { components: { AddressForm } } = this.props;
    return (
      <AddressForm
        ref={(el) => {
          this._addressForm = el;
        }}
      />
    );
  }

  render() {
    const { components: { Button } } = this.props;
    return (
      <Fragment>
        {this.renderAddressForm()}
        <Button>Save</Button>
      </Fragment>
    );
  }
}

export default withComponents(AddressBook);
