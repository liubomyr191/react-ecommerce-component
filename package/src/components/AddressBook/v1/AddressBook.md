### Overview
#### Usage

```jsx
const account = {
  addressBook: [
    {
 address1: "7742 Hwy 23",
 address2: "",
 country: "US",
 city: "Belle Chasse",
 firstName: "Salvos",
 lastName: "Seafood",
 postal: "70037",
 region: "LA",
 phone: "(504) 393-7303"
},
{
 address1: "35 Akin Adesola St",
 address2: "",
 country: "NG",
 city: "Lagos",
 firstName: "Ocean Basket",
 lastName: "Victoria Island",
 postal: "101241",
 region: "Victoria Island",
 phone: "234 816 059 1821"
}
  ]
};

<AddressBook account={account} />
```
