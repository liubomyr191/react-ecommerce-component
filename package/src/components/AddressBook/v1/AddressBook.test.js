import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import AddressBook from "./AddressBook";

test("basic snapshot", () => {
  const component = renderer.create(<AddressBook />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
