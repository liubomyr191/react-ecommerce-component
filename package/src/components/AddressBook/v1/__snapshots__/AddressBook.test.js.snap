// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
Array [
  "AddressForm(undefined)",
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-bottom: 16px;
  padding-left: 0;
  padding-right: 0;
  padding-top: 16px;
}

.c0 > div:last-of-type {
  margin-left: 16px;
}

<div
    className="c0"
  >
    
    Button(undefined)
  </div>,
]
`;
